flink-python/pyflink/common/typeinfo.py

```diff
@@ -160,227 +193,193 @@ class SqlTimeTypeInfo(TypeInformation, ABC):
+        elif self._element_type == Types.LONG():
+            return JBasicArrayTypeInfo.LONG_ARRAY_TYPE_INFO
+        elif self._element_type == Types.FLOAT():
+            return JBasicArrayTypeInfo.FLOAT_ARRAY_TYPE_INFO
+        elif self._element_type == Types.DOUBLE():
+            return JBasicArrayTypeInfo.DOUBLE_ARRAY_TYPE_INFO
+        elif self._element_type == Types.CHAR():
+            return JBasicArrayTypeInfo.CHAR_ARRAY_TYPE_INFO
+        elif self._element_type == Types.STRING():
+            return JBasicArrayTypeInfo.STRING_ARRAY_TYPE_INFO
+        else:
+            raise TypeError("Invalid element type for a primitive array.")
 
-    @staticmethod
-    def FLOAT_ARRAY_TYPE_INFO():
-        return WrapperTypeInfo(
-            get_gateway().jvm.org.apache.flink.api.common.typeinfo
-            .BasicArrayTypeInfo.FLOAT_ARRAY_TYPE_INFO)
+    def __eq__(self, o) -> bool:
+        if isinstance(o, BasicArrayTypeInformation):
+            return self._element_type == o._element_type
+        return False
 
-    @staticmethod
-    def DOUBLE_ARRAY_TYPE_INFO():
-        return WrapperTypeInfo(
-            get_gateway().jvm.org.apache.flink.api.common.typeinfo
-            .BasicArrayTypeInfo.DOUBLE_ARRAY_TYPE_INFO)
+    def __repr__(self):
+        return "BasicArrayTypeInformation<%s>" % self._element_type
 
-    @staticmethod
-    def CHAR_ARRAY_TYPE_INFO():
-        return WrapperTypeInfo(
-            get_gateway().jvm.org.apache.flink.api.common.typeinfo
-            .BasicArrayTypeInfo.CHAR_ARRAY_TYPE_INFO)
 
-    @staticmethod
-    def STRING_ARRAY_TYPE_INFO():
-        return WrapperTypeInfo(
-            get_gateway().jvm.org.apache.flink.api.common.typeinfo
-            .BasicArrayTypeInfo.STRING_ARRAY_TYPE_INFO)
+def is_basic_array_type_info(type_info: TypeInformation):
+    return isinstance(type_info, BasicArrayTypeInformation)
 
 
-def is_basic_array_type_info(type_info: TypeInformation):
-    return type_info in {
-        BasicArrayTypeInfo.BOOLEAN_ARRAY_TYPE_INFO(),
-        BasicArrayTypeInfo.BYTE_ARRAY_TYPE_INFO(),
-        BasicArrayTypeInfo.SHORT_ARRAY_TYPE_INFO(),
-        BasicArrayTypeInfo.INT_ARRAY_TYPE_INFO(),
-        BasicArrayTypeInfo.LONG_ARRAY_TYPE_INFO(),
-        BasicArrayTypeInfo.FLOAT_ARRAY_TYPE_INFO(),
-        BasicArrayTypeInfo.DOUBLE_ARRAY_TYPE_INFO(),
-        BasicArrayTypeInfo.CHAR_ARRAY_TYPE_INFO(),
-        BasicArrayTypeInfo.STRING_ARRAY_TYPE_INFO()
-    }
-
-
-class PickledBytesTypeInfo(WrapperTypeInfo, ABC):
+class PickledBytesTypeInfo(TypeInformation, ABC):
     """
     A PickledBytesTypeInfo indicates the data is a primitive byte array generated by pickle
     serializer.
     """
 
     @staticmethod
     def PICKLED_BYTE_ARRAY_TYPE_INFO():
-        return WrapperTypeInfo(get_gateway().jvm.org.apache.flink.streaming.api.typeinfo.python
-                               .PickledByteArrayTypeInfo.PICKLED_BYTE_ARRAY_TYPE_INFO)
+        return PickledBytesTypeInfo()
```

Comment:
Why do we need to provide this static factory method without parameters? I guess you want to implement a singleton, but doing so will not achieve the effect